### PR TITLE
Wav2vec2.0 Script add

### PR DESCRIPTION
**Pull request by tbartley94**
# What does this PR do ?

Adds a script to load Wav2Vec2.0 weights from Fair to NeMo implementation. Also adjusts NeMo implementation to be similar to Fair's.

**Collection**: [ASR]

# Changelog 
- Added script to download and load state_dict from Fairseq Wav2Vec models to NeMo equivalents
- Adjusted order of normalization operations for wav2vec preprocessing
- Changed Wav2Vec module config so that projection operation of features is done in transformer encoder
- Altered contrastive loss so config can provide option to not combine groups. (Just added an optional boolean)

# Before your PR is "Ready for review"
**Pre checks**:
- [y ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ n] Did you write any new necessary tests?
- [ n] Did you add or update any necessary documentation?
- [ n] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [y ] New Feature
- [ y] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by tbartley94** [comment]
@sam1373 
I can take the script out if it's not necessary or want to expand it to also include the CTC models once those are checked out. But the changes to the Wav2Vec modules should address the discrepancies. I also don't really have much way to test this beyond simply the dict_loading being successful under `strict=True` conditions.

**Comment by sam1373** [comment]
Ideally we'd want to add the ctc models to be able to test the wer, or at least try fine-tuning the pre-trained checkpoints to see that we get reasonable results, otherwise we don't really know if it works.

**Comment by tbartley94** [comment]
Okay, I'll move to draft until then.
